### PR TITLE
[PIL-1155] Add lint and prePrChecks command

### PR DESCRIPTION
**Pull request by ridow-hmrc**
Add lint and prePrChecks command. This also runs scalafmt, which has loads of changes

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -3,16 +3,16 @@ import sbt.*
...
+    "org.slf4j" % "slf4j-simple" % "1.7.36",
+    "org.scalatest" %% "scalatest" % "3.2.17",
+    "org.scalatestplus" %% "selenium-4-12" % "3.2.17.0",
+    "com.vladsch.flexmark" % "flexmark-all" % "0.62.2",
+    "io.cucumber" %% "cucumber-scala" % "8.20.0",
+    "io.cucumber" % "cucumber-junit" % "7.15.0",
+    "junit" % "junit" % "4.13.2",
+    "com.novocode" % "junit-interface" % "0.11",
+    "com.typesafe" % "config" % "1.4.2"
   ).map(_ % Test)
```
Comment (Prashant-digital) [inline project/Dependencies.scala:17]: Formatting has changed the reading easiness. Can we use the formatting which is easy to read?

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/PageObject.scala [hunk 2]
```diff
@@ -19,14 +19,15 @@ import com.typesafe.config.{Config, ConfigFactory}
...
+  val config: Config = ConfigFactory.load()
+  val env: String = config.getString("environment")
+  val defaultConfig: Config = config.getConfig("local")
+  val envConfig: Config = config.getConfig(env).withFallback(defaultConfig)
+  val rootUrl: String = TestConfiguration.url("pillar2-frontend")
+  val p2SubRootUrl: String =
+    TestConfiguration.url("pillar2-submission-frontend")
+  val back = "backLink"
+  val submit = "submit"
   val url: String
```
Comment (Prashant-digital) [inline src/test/scala/uk/gov/hmrc/test/ui/cucumber/PageObject.scala:32]: Formatting has changed the reading easiness. Can we use the formatting which is easy to read?

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Pillar2SubmissionPage.scala [hunk 3]
```diff
@@ -24,35 +24,40 @@ import uk.gov.hmrc.test.ui.driver.BrowserDriver
...
+      case "P2 Submission Agent affinity KB page" => P2SubAgentAffinityKBPage
+      case "P2 Submission Assistant role KB page" => P2SubAssistantRoleKBPage
+      case "P2 Submission Individual affinity KB page" =>
+        P2SubIndividualAffinityKBPage
+      case "P2 Submission Set Up Page"     => P2SubmissionSetUpPage
+      case "BTN Return Submission KB Page" => P2BTNReturnSubmissionKBPage
+      case "P2 ASA Dashboard page"         => P2ASADashboardPage
+      case "P2 ASA Home Page"              => P2ASAHomePage
+      case "P2 Home page"                  => P2HomePage
     }
```
Comment (Prashant-digital) [inline src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Pillar2SubmissionPage.scala:61]: Formatting has changed the reading easiness. Can we use the formatting which is easy to read?

File: src/test/scala/uk/gov/hmrc/test/ui/pages/AuthLoginPage.scala [hunk 4]
```diff
@@ -21,46 +21,67 @@ import uk.gov.hmrc.test.ui.conf.TestConfiguration
...
+  val rfmUrl: String =
+    s"$rootUrl" + "replace-filing-member/security/enter-pillar2-id"
+  val enrolmentKeyField: String = "enrolment[0].name"
+  val identifierNameField: String = "input-0-0-name"
+  val identifierValueField: String = "input-0-0-value"
+  val delegatedEnrolmentKeyField: String = "delegatedEnrolment[0].key"
+  val delegatedIdentifierNameField: String = "input-delegated-0-0-name"
+  val delegatedIdentifierValueField: String = "input-delegated-0-0-value"
+  val delegatedAuthRuleField: String = "delegatedEnrolment[0].delegatedAuthRule"
+  val addDelegatedEnrolmentCTA: String = "[onclick='addDelegatedEnrolment()']"
```
Comment (Prashant-digital) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/AuthLoginPage.scala:85]: Formatting has changed the reading easiness. Can we use the formatting which is easy to read?

**Comment by srikanthhmrc** [comment]
I agree with @Prashant-digital

**Pull request closed without merging**